### ISSUE TITLE
How does LagFeatures & WindowFeatures produce values on unseen data without learning any parameters?

### ISSUE THREAD

**Issue by kylegilde** [feature-engine/feature_engine#465]
I was expecting these transformers to learn some parameters that they can use on unseen data.

**Comment by solegalli** [comment]
which parameters?

**Comment by solegalli** [comment]
LagFeatures just moves values in the past forward, and WindowFeatures takes a mathematical function of a window of values in the past.

They are used for time series forecasting, where past values inform future outcomes.

More in the user guide: https://feature-engine.readthedocs.io/en/1.3.x/user_guide/timeseries/forecasting/index.html

**Comment by kylegilde** [comment]
I guess that I was hoping to use this class in the following use case:

A new date occurs in the test dataset that wasn't in the training dataset. The class would know that it can fill in `ambient_temp_lag_1` with a 9 instead of NaN.

```python
import pandas as pd
from feature_engine.timeseries.forecasting import LagFeatures

X = pd.DataFrame({"ambient_temp": range(10)},
                index=pd.date_range("2020-05-15", periods=10, freq="1d"))

lag_f = LagFeatures(periods=1)

lag_f.fit_transform(X)

X_test = pd.DataFrame({"ambient_temp": 10},
                      index=[X.index.max() + pd.Timedelta(1, 'd')])

lag_f.transform(X_test)
```
```
               ambient_temp	 ambient_temp_lag_1
2020-05-25	10	                 NaN

**Comment by solegalli** [comment]
I see what you mean.

At this stage, you'd have to pass the transformer enough past data to create the features that you need for your forecasting horizon. So if the point you need for index=[X.index.max() + pd.Timedelta(1, 'd') is the one in index=[X.index.max() , your test set needs to have both time stamps, index=[X.index.max() + pd.Timedelta(1, 'd') and index=[X.index.max() . 

How to implement feature creation in forecasting is a common challenge. Different libraries are trying different things with different degrees of success. You may want to check the packages sktime or dart if you haven't done so already. I recommend the first one.

Having said this, with my colleague @KishManani we are trying to come up with the best way to solve this issue. More on this, sometime later this year.

Cheers

**Comment by kylegilde** [comment]
sounds good! Thank you for the recommendations!
